src/main/scala/edu/berkeley/ce/sparkrocks/RockSlicer.scala

```diff
@@ -36,26 +36,28 @@ object RockSlicer {
     val generatedInput = JointGenerator(globalOrigin, boundingBox, rockVolumeInputs, jointSetInputs)
     val starterBlocks = Seq(Block(globalOrigin, generatedInput.rockVolume))
 
+    val startTime = System.nanoTime()
     // Generate a list of initial blocks before RDD-ifying it
-    val (seedJoints, remainingJoints) = if (arguments.numProcessors > 1) {
+    val (seedBlocks, remainingJoints) = if (arguments.numPartitions > 1) {
       // Check if at least one of the input joint sets is persistent
       if (jointSetInputs.forall(_.persistence != 100.0)) {
         println("ERROR: Input joint sets must contain at least one persistent joint set to run in parallel. Rerun" +
           "analysis in serial if all joint sets are non-persistent.")
         System.exit(-1)
       }
 
-      SeedJointSelector.searchJointSets(generatedInput.jointSets, starterBlocks.head, arguments.numProcessors)
+      SeedJointSelector.generateSeedBlocks(generatedInput.jointSets, starterBlocks.head, arguments.numPartitions)
     } else {
-      (Seq.empty[Joint], generatedInput.jointSets.flatten)
+      (starterBlocks, generatedInput.jointSets.flatten)
     }
 
-    val seedBlocks = if (seedJoints.isEmpty) {
-      starterBlocks
-    } else {
-      seedJoints.foldLeft(starterBlocks) { (currentBlocks, joint) =>
-        currentBlocks.flatMap(_.cut(joint))
-      }
+    // If running without forcePartition flag, check if specified number of partitions was found
+    if (!arguments.forcePartition && seedBlocks.length != arguments.numPartitions) {
```

Comment:
I could see two possibilities. One is to treat the partitions argument more like a minimum requirement, so if we can generate `n` or more partitions, we proceed normally, otherwise we prompt the user to consider using the force flag.

An alternative would be to set some range, something like requested partitions plus or minus 10%, and proceed normally if the generated number of partitions is within this range, otherwise prompt the user.

I could see arguments for either. As long as we make it clear to the user what's going on, feel free to go with either one.